CMakeLists.txt

```diff
@@ -87,7 +87,7 @@ endif()
 # Example 'v1.0.0'. They MUST not contain a pre-release label such as '-beta'.
 set(FIRMWARE_VERSION "v4.3.0")
 set(FIRMWARE_BTC_ONLY_VERSION "v4.3.0")
-set(BOOTLOADER_VERSION "v1.0.1")
+set(BOOTLOADER_VERSION "v2.0.0")
```

Comment:
wasn't sure about this one; technically the api is not compatible so I bumped the major version, even though all clients continue to work without any change. :man_shrugging:

Reply:
This seems like a bugfix that isn't breaking backwards compatibility. So I would bump the patch number. Any client "following" the old length-number would be reading out of bounds right?

Reply:
It's an edge case; purely technically it is a backwards incompatible change. Just that our existing clients are not parsing the responses strictly enough for it to matter in practice (they could check for the expected length and raise an error on mismatch, in which case the new clients would actually break without a version check).

In the end, major or minor does not have any real impact or consequence, does it?

the third option is just add a comment to the buggy line and state why it is the wrong length, without changing it, as it doesn't have any real impact.

Reply:
It still seems to me that undocumented features are not part of the API. Bumping the bootloader version will enforce all clients to have two code paths which seems like a maintenance burden.

Actually reading the documentation it seems like the bug is that bootloader version is missing and that the other fields have an invalid offset. If you want to bump the version, why don't you add the bootloader version to the command?